packages/List/src/ListItem.tsx

```diff
@@ -0,0 +1,182 @@
+import * as React from 'react';
+import cx from 'classnames';
+
+import Checkmark from '@igloo-ui/icons/dist/Checkmark';
+import CrownSolid from '@igloo-ui/icons/dist/CrownSolid';
+import UserSolid from '@igloo-ui/icons/dist/UserSolid';
+
+import { VisualIdentifier } from '../../../shared/components';
+
+import './list-item.scss';
+
+interface ListItem {
+  /** Add a colored square instead of an image or an icon */
+  color?: string;
+  /** Icon displayed in front of the option label. */
+  icon?: React.ReactElement;
+  /** Specifies the url for the image to show */
+  src?: string;
+  /** The option value. */
+  value: string | number;
+}
+
+export interface Option extends ListItem {
+  /** Descriptive text to display below the label */
+  description?: string;
+  /** Whether or not the option is disabled */
+  disabled?: boolean;
+  /** The option label. */
+  label: string;
+  /** Display an icon to indicate it's a premium option */
+  premium?: boolean;
+  /** The option type */
+  type: 'list';
+}
+
+export interface Member extends ListItem {
+  /** Displays an icon that shows the member is a manager */
+  manager?: boolean;
+  /** The name of the member */
+  member: string;
+  /** The name of the role display below the member name */
+  role?: string;
+  /** The option type */
+  type: 'member';
+}
+
+export type OptionType = Option | Member;
+
+export interface ListItemProps extends React.ComponentProps<'li'> {
+  /** Add a specific class to the Select. */
+  className?: string;
+  /** True for a compact appearance. */
+  isCompact?: boolean;
+  /** If the option is focused/hovered */
+  isFocused?: boolean;
+  /** If the option is selected */
+  isSelected?: boolean;
```

Comment:
The what is the selected state? The hover state is a blue background. Selected needs to be different, otherwise it would be too confusing for keyboard focus